mmcls/models/necks/reduction.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import copy
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import build_activation_layer, build_norm_layer
+from mmengine.model.utils import initialize
+
+from mmcls.registry import MODELS
+
+
+@MODELS.register_module()
+class LinearReduction(nn.Module):
```

Comment:
```
from mmengine.model import BaseModule
```

and just inherit the BaseModule. then you can just remove the function of `init_weights(self)` and  ```self.init_cfg = copy.deepcopy(init_cfg)```

update the super init line:

```
super(LinearReduction, self).__init__(init_cfg=init_cfg)
```